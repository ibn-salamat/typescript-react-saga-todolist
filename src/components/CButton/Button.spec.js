import { shallow, render, mount } from "enzyme";
import { CButton } from "./index";

const setUp = (props) => shallow(<CButton {...props} />);

describe("Button", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should render component", () => {
    expect(component).toMatchSnapshot();
  });

  it("should call onClick method", () => {
    const mockCallback = jest.fn();
    const Button = setUp({ onClick: mockCallback });

    expect(mockCallback.mock.calls.length).toBe(0);

    Button.find("button").simulate("click");
    Button.find("button").simulate("click");

    expect(mockCallback.mock.calls.length).toBe(2);

    

  });
});
