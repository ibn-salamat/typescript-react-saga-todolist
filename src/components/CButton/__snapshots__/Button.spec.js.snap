// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render component 1`] = `
<div
  className="btn"
>
  <button
    disabled={false}
    onClick={[Function]}
  />
</div>
`;
